packages/commonwealth/server/util/tokenBalanceCache/providers/get_cw721_balances.ts

```diff
@@ -0,0 +1,77 @@
+import { WasmExtension, setupWasmExtension } from '@cosmjs/cosmwasm-stargate';
+
+import { QueryClient } from '@cosmjs/stargate';
+import { factory, formatFilename } from 'common-common/src/logging';
+import { ChainNodeInstance } from '../../../models/chain_node';
+import { rollbar } from '../../rollbar';
+import { Balances } from '../types';
+import { getTendermintClient } from '../util';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+export type GetCw721BalancesOptions = {
+  chainNode: ChainNodeInstance;
+  addresses: string[];
+  contractAddress: string;
+  batchSize?: number;
+};
+
+export async function __getCw721Balances(
+  options: GetCw721BalancesOptions,
+): Promise<Balances> {
+  if (options.addresses.length === 0) return {};
+  const tmClient = await getTendermintClient(options);
+
+  const api = QueryClient.withExtensions(tmClient, setupWasmExtension);
+  const addressesToQuery =
+    options.addresses.length > 1 ? options.addresses : [options.addresses[0]];
+
+  return await getOffChainBatchCw721Balances(
```

Comment:
There is now. I removed it at first because I thought it was just duplicating code, but now I see there is a reason for handling single address queries differently.